lib/networking/api_response.dart

```diff
@@ -0,0 +1,15 @@
+class ApiResponse<T> {
```

Comment:
1. rename the file because it also has the Status enum, not just the response model.
2. this is your response model, not your actual response, so rename both the class as well as the file name appropriately